blocks/api/test/parser.js

```diff
@@ -26,6 +27,25 @@ import {
 import { createBlock } from '../factory';
 import serialize from '../serializer';
 
+/**
+ * Parses the post content with a gremmar pre-generated from PegJS and returns a list of blocks.
+ * It's otherwise identical to parseWithGrammar from parser.js
+ * TODO: find a better way to avoid re-defining this function.
+ *
+ * @param {string} content The post content.
+ *
+ * @return {Array} Block list.
+ */
+export function parsePreGenerated( content ) {
```

Comment:
We can use the following code to avoid duplication in the `blocks/api/parsers.js`:
```js
/**
 * Creates a parse implementation for the post content which returns a list of blocks.
 *
 * @param {Function} parseImplementation Parse implementation.
 *
 * @return {Function} An implementation which parses the post content.
 */
export const createParse = ( parseImplementation ) =>
    ( content ) => parseImplementation( content ).reduce( ( memo, blockNode ) => {
        const block = createBlockWithFallback( blockNode );
        if ( block ) {
            memo.push( block );
        }
        return memo;
    }, [] );

/**
 * Parses the post content with a PegJS grammar and returns a list of blocks.
 *
 * @param {string} content The post content.
 *
 * @return {Array} Block list.
 */
export const parseWithGrammar = createParse( grammarParse );
```